test/IdRegistry/IdRegistry.t.sol

```diff
@@ -300,6 +300,75 @@ contract IdRegistryTest is IdRegistryTestSuite {
         assertEq(idRegistry.getRecoveryOf(1), address(0));
     }
 
+    /*//////////////////////////////////////////////////////////////
+                         VERIFY FID CUSTODY ADDRESS TESTS
+    //////////////////////////////////////////////////////////////*/
+
+    function testFuzzVerifyFidSignature(uint256 recipientPk, bytes32 digest) public {
+        recipientPk = _boundPk(recipientPk);
+        address recipient = vm.addr(recipientPk);
+        bytes memory sig = _signRegister(recipientPk, recipient, address(0), 10);
+
+        vm.prank(owner);
+        idRegistry.disableTrustedOnly();
+        idRegistry.registerFor(recipient, address(0), 10, sig);
+        assertEq(idRegistry.getIdCounter(), 1);
+
+        bytes memory msg_sig = _signDigest(recipientPk, digest);
+        assertEq(idRegistry.verifyFidSignature(recipient, 1, digest, msg_sig), true);
+    }
+
+    function testFuzzCannotVerifyFidSignatureIfBadDigest(
+        uint256 recipientPk,
+        bytes32 digest,
+        bytes32 badDigest
+    ) public {
+        assert(digest != badDigest);
+        recipientPk = _boundPk(recipientPk);
+        address recipient = vm.addr(recipientPk);
+        bytes memory sig = _signRegister(recipientPk, recipient, address(0), 10);
+
+        vm.prank(owner);
+        idRegistry.disableTrustedOnly();
+        idRegistry.registerFor(recipient, address(0), 10, sig);
+        assertEq(idRegistry.getIdCounter(), 1);
+
+        bytes memory msgSig = _signDigest(recipientPk, digest);
+        assertEq(idRegistry.verifyFidSignature(recipient, 1, badDigest, msgSig), false);
+    }
+
+    function testFuzzCannotVerifyFidSignatureIfBadFid(uint256 recipientPk, bytes32 digest) public {
+        recipientPk = _boundPk(recipientPk);
+        address recipient = vm.addr(recipientPk);
+        bytes memory sig = _signRegister(recipientPk, recipient, address(0), 10);
+
+        vm.prank(owner);
+        idRegistry.disableTrustedOnly();
+        idRegistry.registerFor(recipient, address(0), 10, sig);
+        assertEq(idRegistry.getIdCounter(), 1);
+
+        bytes memory msgSig = _signDigest(recipientPk, digest);
+        assertEq(idRegistry.verifyFidSignature(recipient, 2, digest, msgSig), false);
+    }
+
+    function testFuzzCannotVerifyFidSignatureIfBadCustodyAddress(
+        uint256 recipientPk,
+        bytes32 digest,
+        address badCustodyAddress
```

Comment:
is it possible that badCustodyAddress collides with the public key derived from recipientPk? 

if so, we should probably vm.assume that case out.

Reply:
Still getting used to Foundry. Much better with Hardhat.

I added vm.assume where it makes sense in the added tests.